doc/Language/classtut.pod6

```diff
@@ -5,86 +5,60 @@
 
-Raku has a rich built-in syntax for defining and using classes.
+=head1 A quick overview
 
-A default constructor allows the setting of attributes for the created
-object:
+Let's start with an example to give an overview:
 
 =begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-}
-
 class Rectangle {
-    has Point $.lower;
-    has Point $.upper;
+    has Int $.a = 1;
+    has Int $.b = 1;
+
+    submethod TWEAK(:$log) {
+        say "Constructing Rectangle ($!a|$!b)" if $log;
+    }
 
     method area(--> Int) {
-        ($!upper.x - $!lower.x) * ( $!upper.y - $!lower.y);
+        return $!a * $!b;
     }
 }
 
-# Create a new Rectangle from two Points
-my $r = Rectangle.new(lower => Point.new(x => 0, y => 0),
-                      upper => Point.new(x => 10, y => 10));
+my $r1 = Rectangle.new(a => 2, b => 3);
+say $r1.area(); # OUTPUT: «6␤»
 
-say $r.area(); # OUTPUT: «100␤»
+my $r2 = Rectangle.new(b => 2, :log); # OUTPUT: «Constructing Rectangle (1|2)␤»
 =end code
 
-In the two classes, the default constructor is being used. This
-constructor will use named parameters in its invocation: C«Point.new(x =>
-0, y => 0)».
+We define a new I<Rectangle> class using the L<class|/language/objects#Classes> keyword. It has two I<attributes>, C<$!a> and C<$!b> introduced with the C<has> keyword. Both default to C<1>. Read only accessor methods are automatically generated. (Note the C<.> instead of C<!> in the declaration, which triggers the generation. Mnemonic: C<!> resembles a closed door, C<.> an open one.) 
 
-You can also provide your own constructor and C<BUILD> implementation.
-You need to do it for cases in which there are private attributes that
-might need to be populated during object construction, as in the
-example below:
+The L<method|/language/objects#Methods> named C<area> will return the area of the rectangle.
 
-=begin code
-# Example taken from
-# https://medium.freecodecamp.org/a-short-overview-of-object-oriented-software-design-c7aa0a622c83
-class Hero {
-    has @!inventory;
-    has Str $.name;
-    submethod BUILD( :$name, :@inventory ) {
-        $!name = $name;
-        @!inventory = @inventory
-    }
+Using the C<TWEAK> submethod, which is automatically called in the construction process we optionally log that a rectangle was created.
 
-    method act {
-        return @!inventory.pick;
-    }
-}
-my $hero = Hero.new(:name('Þor'),
-                    :inventory(['Mjölnir','Chariot','Bilskirnir']));
-say $hero.act;
-=end code
+It is rarely necessary to explicitly write a constructor. An automatically inherited default constructor called L<new|/language/objects#Object_construction> will automatically initialize attributes from named parameters passed to the constructor.
 
-In this case, we I<encapsulate> the private attribute C<@!inventory>; but
-private instance variables cannot be set by the default constructor, which is
-why we add a C<BUILD> submethod that takes care of that.
+=head1 The Task example
```

Comment:
Also comment made in the previous one, please [check it out](https://github.com/Raku/doc/pull/3794#discussion_r571381038)

Reply:
I'm unsure about the header policy. Are all headers turned into index entries? Is there no way to prevent that? This header makes no sense in the index. But it makes a lot of sense locally on this page and I have no good idea of how to change it without degrading readability. The Task example is the one example referred to in the biggest part of the tutorial. Some header that says: "This is the example that most of the stuff you read below refers to, click here to see it again" is important. It also gives structure as one already sees in the overview that there is an important example. Do you have an idea of how to net-improve the situation?

Reply:
I just got the info, that headers are not by default included in the search index. So I'm now unsure what the issue with this header is. Can you clarify?